cpp/include/kvikio/utils.hpp

```diff
@@ -300,6 +302,7 @@ struct libkvikio_domain {
 
 // Macro overloads of KVIKIO_NVTX_FUNC_RANGE
 #define KVIKIO_NVTX_FUNC_RANGE_1() NVTX3_FUNC_RANGE_IN(libkvikio_domain)
+#ifdef KVIKIO_CUDA_FOUND
```

Comment:
Why is this guard only needed for `KVIKIO_NVTX_FUNC_RANGE_2` but not `KVIKIO_NVTX_FUNC_RANGE_1`?

Reply:
Good catch, it worked because `BASIC_NO_CUDA_EXAMPLE` never materialize `KVIKIO_NVTX_FUNC_RANGE_1`. 
I have included all of them in the ifdef guard now: 168dd0c3f91449a16aadbf2cd84c1fcfbedfadc7